### PR TITLE
Xyl/baddbmm

### PR DESCRIPTION
**Pull request by xue-yun-liang**
Thanks for your contribution and we appreciate it a lot. The following instructions would make your pull request more healthy and more easily get feedback. If you do not understand some items, don't worry, just make the pull request and seek help from maintainers.

## Description

add the baddbmm op

## Checklist
**Before PR**:

- [ ] I have read and followed the workflow indicated in the [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) to create this PR.
- [ ] Pre-commit or linting tools indicated in [Contributors.md](https://github.com/DeepLink-org/DIOPI/blob/main/Contributors.md) are used to fix the potential lint issues.
- [ ] Bug fixes are covered by unit tests, the case that causes the bug should be added in the unit tests.
- [ ] New functionalities are covered by complete unit tests. If not, please add more unit test to ensure the correctness.
- [ ] The documentation has been modified accordingly, including docstring or example tutorials.

**After PR**:

- [ ] CLA has been signed and all committers have signed the CLA in this PR.

### CODE REVIEW

File: impl/ascend/functions/baddbmm.cpp [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+
+    diopiAdd(ctx, outCopy, diopiAlphaMulTensor, diopiBateMulTensor, &other);
+    diopiCastDtype(ctx, out, outCopy);
+
+    return diopiSuccess;
+}
+
+DIOPI_API diopiError_t diopiBaddbmmInp(diopiContextHandle_t ctx, diopiTensorHandle_t input, diopiConstTensorHandle_t batch1, diopiConstTensorHandle_t batch2,
+                                       double beta, double alpha) {
+    diopiBaddbmm(ctx, input, input, batch1, batch2, beta, alpha);
```
Comment (hellozmz) [inline impl/ascend/functions/baddbmm.cpp:96]: 修改成 “return diopiBaddbmm(ctx, input, input, batch1, batch2, beta, alpha);” 好一点

File: impl/ascend/functions/baddbmm.cpp [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    diopiDtype_t outDtype;
+    diopiGetTensorDtype(out, &outDtype);
+
+    // adjust the input's and output's data type
+    diopiDtype_t execType = diopi_dtype_float32;
+
+    diopiTensorHandle_t inCopy;
+    makeTensorLike(ctx, &inCopy, input, execType);
+    diopiCastDtype(ctx, inCopy, input);
+    AscendTensor as_inCopy(inCopy);
```
Comment (hellozmz) [inline impl/ascend/functions/baddbmm.cpp:27]: 使用驼峰式变量名

File: impl/ascend/functions/baddbmm.cpp [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+    diopiCastDtype(ctx, batch1Copy, batch1);
+
+    diopiTensorHandle_t batch2Copy;
+    makeTensorLike(ctx, &batch2Copy, batch2, execType);
+    diopiCastDtype(ctx, batch2Copy, batch2);
+
+    // get the size of batch1 * batch2
+    AscendTensor asTensor1 = AscendTensor(batch1);
+    AscendTensor asTensor2 = AscendTensor(batch2);
+    std::vector<int64_t> batch1_shape = asTensor1.shape();
```
Comment (hellozmz) [inline impl/ascend/functions/baddbmm.cpp:44]: 尽量使用驼峰式变量名，下面还有几个变量名称也是类似问题

**Review by hellozmz — APPROVED** [review]

File: diopi_test/python/conformance/diopi_configs.py [hunk 4]
```diff
@@ -372,8 +372,8 @@
         interface=["torch"],
         is_inplace=True,
         dtype=[Dtype.float32, Dtype.float16, Dtype.float64],
-        atol=1e-4,
-        rtol=1e-4,
+        atol=5e-2,
```
Comment (yangbofun) [inline diopi_test/python/conformance/diopi_configs.py:375]: 在device_config.py中调整

**Review by yangbofun — APPROVED** [review]

**Pull request merged**